pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -545,7 +593,31 @@ type ClientSpec struct {
 }
 
 type CleanupPolicySpec struct {
-	DeleteDataDirOnHosts string `json:"deleteDataDirOnHosts"`
+	DeleteDataDirOnHosts DeleteDataDirOnHostsProperty `json:"deleteDataDirOnHosts,omitempty"`
+	SanitizeDisk         SanitizeDiskSpec             `json:"sanitizeDisk"`
+}
+
+// DeleteDataDirOnHostsProperty
+type DeleteDataDirOnHostsProperty string
```

Comment:
Using `string` is better than defining types. The indirection makes it much more complicated to read the code IMO. We've largely avoided type definitions for strings in Rook because of that.

Reply:
I think it depends, it makes sense when comparing with types. Also, it gives an impression of robustness as well as a Golang good practice. If we have the proper `.String()` helpers I think it's fine. It just that previously we never have that and we were always `string(type)` to access it (if that's what you are thinking about).